app/Http/Controllers/StoreController.php

```diff
@@ -242,6 +245,33 @@ public function postCheckout()
         return js_view('store.order-create');
     }
 
+    public function postRequestNotification($product_id)
+    {
+        $user = Auth::user();
+        $product = Store\Product::findOrFail($product_id);
+
+        if ($product->inStock()) {
+            return error_popup(trans('store.product.notification-in-stock'));
+        }
+
+        $request = Store\NotificationRequest::get($user->user_id, $product_id);
+
+        if ($request) {
+            return error_popup(trans('store.product.notification-already-requested'));
+        }
+
+        $request = Store\NotificationRequest::create([
+            'user_id' => $user->user_id,
+            'product_id' => $product_id,
+        ]);
+
+        if ($request) {
+            return ['ok' => true];
```

Comment:
how about `ujs_redirect <product url>` and use `data-remote` for the request button/link instead?